File: examfront/src/app/services/category.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from '../helper/helper';

@Injectable({
  providedIn: 'root'
})
export class CategoryService {

  constructor( private _http : HttpClient) { }

  // get the single quiz
  public getCategory(cid){
    return this._http.get(`${baseUrl}/api/category/${cid}`)
  }

  public getCategories(){
    return this._http.get(`${baseUrl}/api/category/`)
  }

  public addCategory(category){
    return this._http.post(`${baseUrl}/api/category/`,category);
  }

  public deleteCategory(cid){
    return this._http.delete(`${baseUrl}/api/category/${cid}`);
  }

  //update Quiz
  public updateCategory(category){
    return this._http.put(`${baseUrl}/api/category/`,category)
  }
}

File: JwtExample3/src/main/java/com/jwt/example/services/CategoryService.java
package com.jwt.example.services;

import java.util.List;
import java.util.Set;

import com.jwt.example.entity.exam.Category;
import com.jwt.example.entity.exam.Quiz;

public interface CategoryService {

    public Category addCategory(Category Category);
    
    public Category updateCategory(Category category);

    public Set<Category> getCategories();

    public Category getCategory(Long categoryId);

    public void deleteCategory(Long categoryId);


}